core/controlplane/config/templates/cloud-config-etcd

```diff
@@ -355,6 +377,43 @@ write_files:
     content: {{$w.GzippedBase64Content}}
   {{- end }}
 {{- end }}
+{{if and (.AmazonSsmAgent.Enabled) (ne .AmazonSsmAgent.DownloadUrl "")}}
+  - path: "/opt/bin/ssm/install-ssm-agent.sh"
+    permissions: 0700
+    content: |
+      #!/bin/bash
+      set -eo pipefail
+
+      function finish {
+        if [ "$(pwd)" == "/opt/bin/ssm" ]; then
+          popd > /dev/null 2>&1
+        fi
+      }
+
+      trap finish EXIT
+
+      pushd /opt/bin/ssm > /dev/null 2>&1
```

Comment:
so executables are in `/opt/bin/ssm/<name>?` It is subjective, but I think `/opt/ssm/bin/<name>` is more natural